util/base1024/decode.go

```diff
@@ -1,42 +1,78 @@
 package base1024
 
+import (
+	"github.com/samber/lo"
+	"strings"
+)
+
 func DecodeString(s string) ([]byte, error) {
-	//tail := false
-	//if strings.HasSuffix(s, string(TAIL)) {
-	//	s = s[0 : len(s)-len(TAIL)]
-	//	//input = input.slice(0, input.length - TAIL.length)
-	//	tail = true
-	//}
+
+	tail := false
+	if strings.HasSuffix(s, TAIL) {
+		s = s[0 : len(s)-len(TAIL)]
+		//input = input.slice(0, input.length - TAIL.length)
+		tail = true
+	}
 
 	// Decode input string
 	//points := Array.from(input).map((emoji: string) => EMOJIS.indexOf(emoji))
+	arr := strings.Split(s, "")
 	points := make([]int, 0)
-	//for _, item := s {
-	//	points = append(points, lo.IndexOf(Emojis, item))
-	//}
 
-	//points := []uint8{1, 2, 4, 5, 6, 2, 53, 33, 4, 234, 24}
-	remainder := len(points) % 4
-	safe := len(points) - remainder
+	for i := 0; i < len(arr); i++ {
+		points = append(points, lo.IndexOf[string](Emojis, arr[i]))
+	}
+
+	pointsLength := len(points)
+	remainder := pointsLength % 4
+	safe := pointsLength - remainder
 	source := make([]int, 0)
-	//const source: number[] = []
+
 	var i int
 	for i = 0; i <= safe; i += 4 {
```

Comment:
All of the `for` seems to work like `lo.Chunk()`

So, combine `lo.Chunk()` with `lo.ForEach()` or `lo.Map()`